### PR TITLE
Olelensmar/feat/values files in subfolders

### PR DESCRIPTION
**Pull request by olensmar**
This PR adds support for values files in subfolders to a helm chart. It also fixes / improves a bunch of other issues:
- Consistent parsing and improved validation of YAML errors
- Show YAML errors in Monaco in read-only mode
- Fixed deletion of Helm Charts
- Disabled "Add Resource" action for Helm / Values files in the File Explorer

## Changes

-

## Fixes

- #1295 
- #1385 
- #1384 

## How to test it

- Load a Helm Chart with values files in a subfolder to the chart itself - these should show up in the Helm pane also

## Screenshots

-

## Checklist

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/services/helm.ts [hunk 1]
```diff
@@ -40,27 +46,53 @@ export function getHelmChartFromFileEntry(fileEntry: FileEntry, helmChartMap: He
...
 
 /**
- * check if the k8sResource is supported
- * @param resource
- * @returns @boolean
+ * check if the k8sResource is supported - currently excludes any files
+ * that seem to contain Helm template or Monokle Vanilla template content
  */
+
 export function isSupportedHelmResource(resource: K8sResource): boolean {
```
Comment (devcatalin) [inline src/redux/services/helm.ts:74]: maybe we should rename this function to just `isSupportedResource` ?

File: src/redux/services/helm.ts [hunk 2]
```diff
@@ -40,27 +46,53 @@ export function getHelmChartFromFileEntry(fileEntry: FileEntry, helmChartMap: He
...
+/**
+ * Adds the values file at the given path to the specified HelmChart
+ */
 
-  return Boolean(resource.text.match(helmVariableRegex)?.length) === false;
+export function createHelmValuesFile(fileEntry: FileEntry, helmChart: HelmChart, helmValuesMap: HelmValuesMapType) {
+  const helmValues: HelmValuesFile = {
+    id: uuidv4(),
+    filePath: fileEntry.filePath,
+    name: fileEntry.filePath.substring(path.dirname(helmChart.filePath).length + 1),
```
Comment (devcatalin) [inline src/redux/services/helm.ts:88]: ins't `path.basename(fileEntry.filePath)` enough here?

File: src/redux/services/resource.ts [hunk 3]
```diff
@@ -657,6 +658,10 @@ export function extractK8sResources(fileContent: string, relativePath: string) {
           documents[docIndex]
         );
       } else {
+        if (doc.warnings.length > 0) {
+          log.warn('Doc has warnings', doc.warnings, doc);
```
Comment (devcatalin) [inline src/redux/services/resource.ts:662]: let's add `[extractK8sResources]:` at the beginning of this warning message

File: src/redux/services/validation.ts [hunk 4]
```diff
@@ -54,6 +54,35 @@ export function validateResource(resource: K8sResource) {
...
+  const doc = parseDocument(resource.text);
+  if (doc.errors.length > 0) {
+    const lines = resource.text.split('\n');
+    if (lines[0] === '---') {
+      lines.shift();
+    }
+
+    errors.push(
+      ...doc.errors.map(err => {
+        const line = err.linePos ? lines[err.linePos[0].line].trim() : '';
```
Comment (devcatalin) [inline src/redux/services/validation.ts:69]: is it possible for `err.linePos[0].line` to be undefined? maybe we should do a sanity check

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/redux/services/helm.ts (hunk 2 above)
Comment (olensmar) [reply src/redux/services/helm.ts:88]: no - this is meant to be relative to the containing HelmChart - so the name would be "subfolder/values.yaml" for a values file in a subfolder - not sure what else we could/should use as the name as just the filename might be confusing if the same name is used in different folders

**Review by devcatalin — APPROVED** [review]

**Pull request merged**